File: 4.semester/IJA/src/application/GetReal.java
/* 
 * Project: 	Implementing Block Editor in JavaFX
 * Course: 		IJA 2018
 * Team name: 	Tym xzubri00
 * Members: 
 *		Liscinsky Matus	xlisci02
 *		Zubrik Tomas 	xzubri00
 */
package application;
public class GetReal extends Block{
	
	/**
	 * Constructor of GetReal block
	 */
	public GetReal() {
		this.name="getReal";

	    Port p1 = new Port();
	    p1.in_block=this;
	    p1.addTypeKey("real");
	    p1.addTypeKey("im");
	    this.addInPort(p1);

	    Port out = new Port();
	    out.addTypeKey("real");
	    out.in_block=this;
	    this.addOutPort(out);
	}

	/**
	 * Function gets real part of complex number and writes it to output port
	 */
	public void Execute() {
		double real = this.inPorts.get(0).m.get("real");
	    this.outPorts.get(0).setReal(real);	
	}
    
}

File: 4.semester/ICP/src/getreal.cpp
/**
* @file getreal.cpp
* @brief GetReal implementation
* @author Matúš Liščinský xlisci02, Marcin Vladimír xmarci10
*/
#include "getreal.h"

/**
 * @brief GetReal constructor
 */
GetReal::GetReal()
{
    this->type=BlockType::gr;

    Port p1 = Port();
    p1.in_block=this;
    p1.addTypeKey("real");
    p1.addTypeKey("im");
    this->addInPort(p1);

    Port out = Port();
    out.in_block=this;
    out.addTypeKey("real");
    this->addOutPort(out);
}

/**
 * @brief Extract real part of given complex number
 */
void GetReal::Execute(){
    double real = this->inPorts[0].m.at("real");
    this->outPorts[0].setReal(real);
}
